e2e flake: Failed to create static ip

https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/22242/kubernetes-pull-build-test-e2e-gce/31096/?debugUI=CLOUD

```
NAME                             NETWORK             SRC_RANGES RULES   SRC_TAGS TARGET_TAGS
e2e-gce-builder-3-3-master-https e2e-gce-builder-3-3 0.0.0.0/0  tcp:443          e2e-gce-builder-3-3-master
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - Internal Error
 
Attempt 1 failed to create static ip e2e-gce-builder-3-3-master-ip. Retrying.
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - The resource 'projects/kubernetes-jenkins-pull/regions/us-central1/addresses/e2e-gce-builder-3-3-master-ip' already exists
 
Attempt 2 failed to create static ip e2e-gce-builder-3-3-master-ip. Retrying.
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - The resource 'projects/kubernetes-jenkins-pull/regions/us-central1/addresses/e2e-gce-builder-3-3-master-ip' already exists
 
Attempt 3 failed to create static ip e2e-gce-builder-3-3-master-ip. Retrying.
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - The resource 'projects/kubernetes-jenkins-pull/regions/us-central1/addresses/e2e-gce-builder-3-3-master-ip' already exists
 
Attempt 4 failed to create static ip e2e-gce-builder-3-3-master-ip. Retrying.
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - The resource 'projects/kubernetes-jenkins-pull/regions/us-central1/addresses/e2e-gce-builder-3-3-master-ip' already exists
 
Attempt 5 failed to create static ip e2e-gce-builder-3-3-master-ip. Retrying.
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - The resource 'projects/kubernetes-jenkins-pull/regions/us-central1/addresses/e2e-gce-builder-3-3-master-ip' already exists
 
Failed to create static ip e2e-gce-builder-3-3-master-ip 
2016/02/29 18:22:44 e2e.go:278: Error running up: exit status 2
2016/02/29 18:22:44 e2e.go:274: Step 'up' finished in 1m49.20839755s
2016/02/29 18:22:44 e2e.go:140: Error starting e2e cluster. Aborting.
exit status 1
Build step 'Execute shell' marked build as failure
```

@ixdy

--- a/cluster/gce/util.sh
+++ b/cluster/gce/util.sh
@@ -298,19 +298,27 @@ function create-static-ip {
   local attempt=0
   local REGION="$2"
   while true; do
-    if ! gcloud compute addresses create "$1" \
+    if gcloud compute addresses create "$1" \
       --project "${PROJECT}" \
       --region "${REGION}" -q > /dev/null; then
-      if (( attempt > 4 )); then
-        echo -e "${color_red}Failed to create static ip $1 ${color_norm}" >&2
-        exit 2
-      fi
-      attempt=$(($attempt+1)) 
-      echo -e "${color_yellow}Attempt $attempt failed to create static ip $1. Retrying.${color_norm}" >&2
-      sleep $(($attempt * 5))
-    else
+      # successful operation
       break
     fi
+
+    if cloud compute addresses describe "$1" \
+      --project "${PROJECT}" \
+      --region "${REGION}" >/dev/null 2>&1; then
+      # it exists - postcondition satisfied
+      break
+    fi
+
+    if (( attempt > 4 )); then
+      echo -e "${color_red}Failed to create static ip $1 ${color_norm}" >&2
+      exit 2
+    fi
+    attempt=$(($attempt+1))
+    echo -e "${color_yellow}Attempt $attempt failed to create static ip $1. Retrying.${color_norm}" >&2
+    sleep $(($attempt * 5))
   done
 }
 

